00_basics/sc_dimensions.ipynb

```diff
@@ -0,0 +1,1059 @@
+   "source": [
+    "A bit unexpecting that we do not receive an error - SuperCollider expands arrays in such operations to the longest array and cycles through the shorter ones. This allows for very concise code, but you have to take care about the input arrays yourself.\n",
+    "Lets see how this cascades when we want to multiply 3 arrays with different size."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "id": "tender-footwear",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "-> [ 56, 120, 108, 80 ]"
+     ]
+    }
+   ],
+   "source": [
+    "[2, 3] * [4, 5, 6] * [7, 8, 9, 10]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "senior-purse",
+   "metadata": {},
+   "source": [
+    "The results are calculated as following:\n",
+    "\n",
+    "$$\n",
+    "\\begin{aligned}\n",
+    "2 \\cdot 4 \\cdot 7 &= 7\\\\\n",
+    "3 \\cdot 5 \\cdot 8 &= 120\\\\\n",
+    "2 \\cdot 6 \\cdot 9 &= 108\\\\\n",
+    "3 \\cdot 4 \\cdot 10 &= 80\n",
+    "\\end{aligned}\n",
+    "$$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "diagnostic-overall",
+   "metadata": {},
+   "source": [
+    "### Matrix\n",
+    "\n",
+    "Now we can also take a look at how a matrix is represented in SuperCollider."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "id": "intellectual-assurance",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "-> [ [ 2, 3, 4 ], [ 4, 5, 6 ] ]"
+     ]
+    }
+   ],
+   "source": [
+    "n = [\n",
+    "    [2, 3, 4],\n",
+    "    [4, 5, 6],\n",
+    "]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "desirable-platform",
+   "metadata": {},
+   "source": [
+    "So it is basically just an array of arrays - but lets take a closer look at this as it is important to get the indices and the dimensions properly when working with matrices.\n",
+    "\n",
+    "Lets say we want to represent a $3 \\times 2$ matrix \n",
+    "$$\n",
+    "\\text{M} =\n",
+    "\\begin{pmatrix}\n",
+    "   1 & 2\\\\\n",
+    "   3 & 4\\\\\n",
+    "   5 & 6\n",
+    "\\end{pmatrix}\n",
+    "$$"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "id": "cooked-beijing",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[ [ 1, 2 ], [ 3, 4 ], [ 5, 6 ] ]\n",
+      "-> 2"
+     ]
+    }
+   ],
+   "source": [
+    "m = [\n",
+    "    [1, 2],\n",
+    "    [3, 4],\n",
+    "    [5, 6],\n",
+    "];\n",
+    "m.postln;\n",
+    "m[0][1];"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "determined-metadata",
+   "metadata": {},
+   "source": [
+    "We see that the first dimension in a SuperCollider array are the rows, the 2nd dimension (the content of the arrays) are the columns - this fits nicely with our mathematical notation.\n",
```

Comment:
We interpret the first dimension as rows, the second dimension (the elements of the arrays) as columns